From 05297aac9168d7a813bc357801fcab7d90b2f5e9 Mon Sep 17 00:00:00 2001
From: Simon Busch <morphis@gravedo.de>
Date: Thu, 30 Oct 2014 13:04:54 +0100
Subject: [PATCH] TextInput: forward passwordCharacter property from TextInput

Depending on the application it might be needed to exchange the password character
and not use the platform default one.

Upstream-Status: pending
---
 src/controls/TextField.qml | 13 +++++++++++++
 1 file changed, 13 insertions(+)

diff --git a/src/controls/TextField.qml b/src/controls/TextField.qml
index 666d137..9751999 100644
--- a/src/controls/TextField.qml
+++ b/src/controls/TextField.qml
@@ -175,6 +175,19 @@ Control {
     Accessible.passwordEdit: echoMode == TextInput.Password || echoMode === TextInput.PasswordEchoOnEdit
 
     /*!
+        \qmlproperty string TextField::passwordCharacter
+
+        This is the character displayed when echoMode is set to Password or
+        PasswordEchoOnEdit. By default it is the password character used by the platform
+        theme.
+
+        If this property is set to a string with more than one character, the first
+        character is used. If the string is empty, the value is ignored and the property
+        is not set.
+    */
+    property alias passwordCharacter: textInput.passwordCharacter
+
+    /*!
         \qmlproperty font TextField::font
 
         Sets the font of the TextField.
-- 
2.1.2

